docs/topics/control-flow.md

```diff
@@ -50,12 +50,58 @@ val max = if (a > b) {
 If you're using `if` as an expression, for example, for returning its value or
 assigning it to a variable, the `else` branch is mandatory.
 
-## When expression
+## When expressions and statements
 
-`when` defines a conditional expression with multiple branches. It is similar to the `switch` statement in C-like languages.
-Its simple form looks like this.
+`when` is a conditional expression that executes code based on multiple possible values or conditions. It is
+similar to the `switch` statement in C-like languages. For example:
```

Comment:
```suggestion
closely related to the `switch` statement in Java, C, and similar languages. For example:
````

I feel like these days "C-like languages" is not as well understood as it was a few years ago.